### PR TITLE
Initial Pseudonymization

### PR DESCRIPTION
**Pull request by koppor**
This adds an initial anonymization of BibTeX libraries.

Use case is to make `.bib` files available inside JabRef's repository for reproducing issues. One thing are performance issues, other things are certain quality checks.

The current implementation is very basic, but enough for me currently.

I would propose to merge this in and refine afterwards. TODOs are inside. Other future work is to include this functionality in the CLI (and the UI). In the UI similar to library-based-on-aux-generation.

NO CHANGELOG entry, because functionality accessible through tests only.

### Mandatory checks

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
You know that we have a bib file generator python script to quickly generate 1000s of entries ? https://github.com/JabRef/jabref/blob/main/scripts/bib-file-generator.py

In addtion, for reproducing issues or so it does not make any sense.

**Comment by koppor** [comment]
> In addtion, for reproducing issues or so it does not make any sense.

I will implement the feature to check a BibTeX file in the context of a paper. Here, I need mappings of real examples. This is (for me) more easy than replicating the properties of the real example in a Python script. I am think of end-to-end tests closer to reality.

I can also include this code in a follow-up pull request.

**Comment by koppor** [comment]
This PR (should) also fix issues with `.bib` files maintained in this repository.

**Comment by koppor** [comment]
Also fixes `equals` for `BibDatabaseContext`. The eventBus object (created for each context) was also compared. Therefore, even if equal at other fields, two context were never equal.

**Comment by koppor** [comment]
The image at PR description at https://github.com/JabRef/jabref/pull/10778 is my use case.

File: src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                return Integer.parseInt(key.substring(key.lastIndexOf('-') + 1));
+            } catch (NumberFormatException e) {
+                return Integer.MAX_VALUE;
+            }
+        }
+    }
+
+    public Result pseudonymizeLibrary(BibDatabaseContext bibDatabaseContext) {
+        // TODO: Anonymize metadata
+        // TODO: Anonymize strings
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java:66]: Followup?

File: src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+                csvPrinter.printRecord("pseudonymized", "original value");
+                valueMapping.entrySet().stream()
+                            // We have date-1, date-2, ..., date-10, date-11. That should be sorted accordingly.
+                            .sorted(Comparator.comparing((Map.Entry<String, String> entry) -> getKeyPrefix(entry.getKey())
+                            ).thenComparingInt(entry -> extractNumber(entry.getKey())))
+                            .forEach(Unchecked.consumer(entry -> {
+                                csvPrinter.printRecord(entry.getKey(), entry.getValue());
+                            }));
+            }
+        }
```
Comment (calixtus) [inline src/main/java/org/jabref/logic/pseudonymization/Pseudonymization.java:48]: Nope. ;-)

**Review by calixtus — APPROVED** [review]

**Pull request merged**